app/services/DuplicateService.scala

```diff
@@ -35,13 +35,31 @@ class DuplicateService @Inject() (
     newLocalReferenceNumber: LocalReferenceNumber
   )(implicit hc: HeaderCarrier): Future[Boolean] = cacheConnector.get(oldLocalReferenceNumber) flatMap {
     case Some(userAnswers) =>
-      val updatedUserAnswers: UserAnswers = userAnswers.copy(lrn = newLocalReferenceNumber, isSubmitted = Some(RejectedPendingChanges))
-      cacheConnector.post(
-        updatedUserAnswers
-      ) // TODO CTCP-3469 Will have to keep any draft declaration with same LRN, can probably handle this when the doesDraftOrSubmissionExistForLrn is called in the backend
+      val updatedUserAnswers: UserAnswers =
+        userAnswers.copy(lrn = newLocalReferenceNumber, isSubmitted = Some(RejectedPendingChanges))
+      cacheConnector
+        .post(
+          updatedUserAnswers
+        )
+        .flatMap {
+          case true  => updateResubmittedLrn(newLocalReferenceNumber, userAnswers)
```

Comment:
(I know I did the cache connector part) but Future[Boolean] is a little bit of an anti-pattern as we already have a monad for handling failures in the future. To refactor this you may want to consider refactoring the connector to be maybe Future[Unit] that fails on 400's / 500's though this may be out of scope for this ticket